.github/actions/docusaurus-swizzled-warning/index.js

```diff
@@ -0,0 +1,135 @@
+'use strict'
+
+const core = require('@actions/core')
+const github = require('@actions/github')
+const fetch = require('node-fetch')
+const {
+  getAllFilesForPullRequest,
+  getChangedFilesBetweenTags,
+  findKeyEndingWith,
+} = require('./helpers')
+
+async function run() {
+  try {
+    const token = core.getInput('github-token', { required: true })
+
+    const { pull_request: pr } = github.context.payload
+    if (!pr) {
+      throw new Error('Event payload missing `pull_request`')
+    }
+
+    const client = github.getOctokit(token)
+    const packageName = 'docusaurus-theme-openapi'
+    const packageParentName = 'docusaurus-preset-openapi'
+    const overideComponents = ['Curl', 'Response']
+    const messageTemplate = `<table><thead><tr><th>
+      ⚠️ This PR contains changes to components of ${packageName} we've overridden
+    </th></tr></thead>
+    <tbody><tr><th>
+      We need to watch out for changes to the ${overideComponents.join(
+        ', ',
+      )}components
+    </th></tr>
+    `
+
+    if (
+      ['dependabot[bot]', 'dependabot-preview[bot]'].includes(pr.user.login)
+    ) {
+      const files = await getAllFilesForPullRequest(
+        client,
+        github.context.repo.owner,
+        github.context.repo.repo,
+        pr.number,
+      )
+
+      for (const file of files) {
+        if (file.filename !== 'package-lock.json') {
+          continue
+        }
+
+        const pkgLockNewJson = await (await fetch(file.raw_url)).json()
+        const pkgLockOldJson = await (
+          await fetch(
+            `https://raw.githubusercontent.com/${github.context.repo.owner}/${github.context.repo.repo}/master/${file.filename}`,
+          )
+        ).json()
```

Comment:
I don't think we need to pull in node-fetch as a dependency and request these URLs. We've got an OctoKit instance in scope, so I think we should be able to use:

```
octokit.rest.repos.getContent({
  owner,
  repo,
  path,
  ref?,
});
```

to replace both these calls.

Reply:
I am new to octokit, I will look into this and add this if applicable

Reply:
solved with https://github.com/badges/shields/pull/9467/commits/db2e03c72368ba629098e9cb1474566b8dfd15ed
notice that files are > 1MB therefor an getContent won't return content (tried even with raw)
I had to use the sha and make a new blob request.